syne_tune/optimizer/schedulers/hyperband.py

```diff
@@ -483,6 +506,25 @@ def __init__(self, config_space: Dict[str, Any], **kwargs):
         # least once, this records the sum of costs for reaching its last
         # recent milestone.
         self._cost_offset = dict()
+        self._initialize_early_checkpoint_removal(
+            kwargs.get("early_checkpoint_removal_kwargs")
+        )
+
+    def _initialize_early_checkpoint_removal(
+        self, callback_kwargs: Optional[Dict[str, Any]]
+    ):
+        if callback_kwargs is not None:
+            for name in ["max_num_checkpoints"]:
+                assert (
+                    name in callback_kwargs
+                ), f"early_checkpoint_removal_kwargs must contain '{name}' entry"
+            callback_kwargs = dict(
+                callback_kwargs,
+                metric=self.metric,
+                resource_attr=self._resource_attr,
+                mode=self.mode,
+            )
+        self._early_checkpoint_removal_kwargs = callback_kwargs
```

Comment:
The CP removal method needs at least `max_num_checkpoints` provided by the user, so just a flag does not work.
I can move all this here into `callback_for_checkpoint_removal`, this would be cleaner.

Reply:
Actually, I think I'd like to keep it the way it is. I need to check that required fieds are contained in `callback_kwargs`, that should be done at construction.

The idea of passing `None` if something is not to be used, is pretty standard.

Reply:
I think we should
1/ check that all required fields are passed by the users at initialization in this function. We really only need to check user-provided params since all other are added by us so we know they are tere
2/ move all kwargs generation into `callback_for_checkpoint_removal`. As you said it would be cleaner this way and the params are already verified by what remains in this funct.

Reply:
OK